### PR TITLE
Bluetooth: Allow use of Characteristic attribute to notify/indicate

### PR DESCRIPTION
**Pull request by Vudentz**
This makes bt_gatt_notify/bt_gatt_indicate easier to use since characteristic value is normally declared with BT_GATT_CHARACTERISTIC.

### CODE REVIEW

File: subsys/bluetooth/host/gatt.c [hunk 1]
```diff
@@ -685,6 +685,18 @@ static int gatt_indicate(struct bt_conn *conn,
 {
 	struct net_buf *buf;
 	struct bt_att_indicate *ind;
+	u16_t value_handle = params->attr->handle;
+
+	/* Check if attribute is a characteristic then adjust the handle */
+	if (bt_uuid_cmp(params->attr->uuid, BT_UUID_GATT_CHRC)) {
```
Comment (jhedberg) [inline subsys/bluetooth/host/gatt.c]: Shouldn't that be `!bt_uuid_cmp()` if you want to check for equality?

File: subsys/bluetooth/host/gatt.c [hunk 2]
```diff
@@ -806,11 +818,23 @@ int bt_gatt_notify(struct bt_conn *conn, const struct bt_gatt_attr *attr,
 		   const void *data, u16_t len)
 {
 	struct notify_data nfy;
+	u16_t value_handle = attr->handle;
 
 	__ASSERT(attr && attr->handle, "invalid parameters\n");
 
+	/* Check if attribute is a characteristic then adjust the handle */
+	if (bt_uuid_cmp(attr->uuid, BT_UUID_GATT_CHRC)) {
```
Comment (jhedberg) [inline subsys/bluetooth/host/gatt.c]: Same here

File: samples/bluetooth/gatt/cts.c [hunk 3]
```diff
@@ -110,5 +110,5 @@ void cts_notify(void)
 	}
 
 	ct_update = 0;
-	bt_gatt_notify(NULL, &attrs[3], &ct, sizeof(ct));
+	bt_gatt_notify(NULL, &attrs[1], &ct, sizeof(ct));
```
Comment (jhedberg) [inline samples/bluetooth/gatt/cts.c]: In all other places the new index is `old - 1` but here it's `old - 2`. Is that correct?

**Review by jhedberg — DISMISSED** [review]

File: samples/bluetooth/gatt/cts.c (hunk 3 above)
Comment (Vudentz) [reply samples/bluetooth/gatt/cts.c]: This was wrong actually, the right index would be 2.

File: include/bluetooth/gatt.h [hunk 4]
```diff
@@ -683,12 +683,13 @@ ssize_t bt_gatt_attr_read_cpf(struct bt_conn *conn,
  *  all peer that have notification enabled via CCC otherwise do a direct
  *  notification only the given connection.
  *
- *  The attribute object must be the so called Characteristic Value Descriptor,
- *  its usually declared with BT_GATT_DESCRIPTOR after BT_GATT_CHARACTERISTIC
- *  and before BT_GATT_CCC.
+ *  The attribute object can be the so called Characteristic Declaration,
+ *  its usually declared with BT_GATT_CHARACTERISTIC followed by BT_GATT_CCC,
```
Comment (carlescufi) [inline include/bluetooth/gatt.h]: * which is usually declared...

**Review by carlescufi — CHANGES_REQUESTED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**